pkg/cli/delete_helm.go

```diff
@@ -184,6 +185,13 @@ func DeleteHelm(ctx context.Context, platformClient platform.Client, options *De
 		}
 	}
 
+	// delete coreDNS components since they're separately deployed and not with the vCluster helm chart
+	err = coredns.DeleteCoreDNSComponents(ctx, cmd.kubeClient, cmd.Namespace)
+	cmd.log.Info("Deleting CoreDNS components...")
+	if err != nil {
+		return fmt.Errorf("delete coreDNS components: %w", err)
```

Comment:
If the cleanup fails on any step later in the process (e.g. `find.ListVClusters`) then you cannot re-run it to clean remaining resources, because it will fail here (because coredns was deleted in the previous run. I think we can consider just warning here instead of erroring out